src/v1/feedback/db.ts

```diff
@@ -0,0 +1,181 @@
+            if (predict.data()) {
+              return { predictedWait: predict.id, data: predict.data() };
+            }
+
+            return [];
+
+          }
+          // get data at all predicted wait times (on specific hour)
+          return this.getHourFeedback(dayRef, hour);
+        }
+
+        // get data at all hours (on specific day)
+        return this.getDayFeedback(eateryRef, day);
+
+      }
+
+      // get data at all days (on specific eatery)
+      return this.getEateryFeedback(ref, eatery);
+
+    }
+
+    // get data at all eateries
+    const eateries = (await ref.get()).docs;
+    const data = await Promise.all(eateries.map(async doc => {
+      const docID = doc.id;
+      const docData = await this.getEateryFeedback(ref, docID);
+      return ({ id: docID, data: docData });
+    }));
+    return data.filter(doc => doc.data.length > 0);
+
+  }
+
+  // initialize docs in firebase
+  async createDocs() {
+
+    for (const eatery in DISPLAY_MAP) {
+      if (Object.prototype.hasOwnProperty.call(DISPLAY_MAP, eatery)) {
+        const eRef = firebaseDB.collection('feedbackData').doc(eatery);
+        eRef.set({});
+        for (const day in this.num_to_day) {
+          if (Object.prototype.hasOwnProperty.call(this.num_to_day, day)) {
+            for (let hour = 0; hour < 24; hour += 1) {
+              const eatRef = eRef.collection(this.num_to_day[day]).doc(hour.toString());
+              eatRef.set({});
+              const eateryRef = eatRef.collection('modelPrediction').doc('0');
+              eateryRef.set(
+                {
+                  observedWait: 0,
+                  count: 0,
+                  comments: []
+                });
+            }
+          }
+        }
+      }
+
+    }
+
+  }
+
+  // POST feedback
+  async addFeedback(feedback: Feedback) {
+
+    const time = new Date();
+    const day = this.num_to_day[time.getDay()];
+    const hour = time.getHours();
+
+    const { eatery } = feedback;
+    const { predictedWait } = feedback;
+    const { observedWait } = feedback;
+    const { comment } = feedback;
+
+    let oldCount = 0;
+    let oldAvg = observedWait;
+
+    // initialized fields
+    let observedWaitAvg = observedWait;
+    let count = 1;
+    let comments = [];
+
+    const eateryRef = firebaseDB
+      .collection('feedbackData')
+      .doc((eatery))
+      .collection(day)
+      .doc(hour.toString())
+      .collection('modelPrediction')
+      .doc(predictedWait.toString());
+
+    // update based on current data
+    // get current data
+    await eateryRef.get()
+      .then(res => {
+        // if doc found, get data
+        oldCount = res.data().count;
+        oldAvg = res.data().observedWait;
+        comments = res.data().comments;
+
+        observedWaitAvg = (oldAvg * oldCount + observedWait) / (oldCount + 1);
+        count = oldCount + 1;
+      })
+      .catch(err => {
+        // else, set initial data
+      })
```

Comment:
Great job on this! Were you doing to call some other function here? If not, it might be better to propagate this error towards the GET route so the client can see the error message. I think you can just throw an error with a custom message.

Reply:
Nope, it was just there to catch an error, but good point! I'll throw an error here, thanks for the suggestion!

Reply:
Nah, I think we should still send some sort of an error message. Even if it's not a message taken straight from the Firebase Error, we should throw an error like `Could not fetch appropriate times for this dining hall` or something along those lines, so in the case that something does get messed up on the database we can easily diagnose it.